### PR TITLE
C# analyzer and formatting

### PR DESCRIPTION
**Pull request by externl**
- Add StyleCop
- Format C# code
- Fix many lints and recommendations

Still various things to fix:
- Setting `<AnalysisMode>All</AnalysisMode>` causes many warnings
- various dotnet format warnings:
```
Msbuild failed when processing the file '/Users/joe/Developer/zeroc-ice/ice-demos/csharp/Ice/callback/msbuild/client/client.csproj' with message: /Users/joe/.nuget/packages/zeroc.icebuilder.msbuild/5.0.9/build/zeroc.icebuilder.msbuild.csharp.targets: (53, 9): Ice Installation invalid or not detected. Invalid IceToolsPath setting `'
...
Unable to fix SA1600. Code fix SettingsFileCodeFixProvider doesn't support Fix All in Solution.
Unable to fix SA1600. Code fix InheritdocCodeFixProvider didn't return a Fix All action.
```

I tried to add dotnet format as a separate workflow but it failed to restore packages.
```
Run dotnet format --verify-no-changes "csharp/C# NET demos.sln"
Unhandled exception: System.Exception: Restore operation failed.
   at Microsoft.CodeAnalysis.Tools.CodeFormatter.OpenMSBuildWorkspaceAsync(String solutionOrProjectPath, WorkspaceType workspaceType, Boolean noRestore, Boolean requiresSemantics, String binaryLogPath, Boolean logWorkspaceWarnings, ILogger logger, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Tools.CodeFormatter.FormatWorkspaceAsync(FormatOptions formatOptions, ILogger logger, CancellationToken cancellationToken, String binaryLogPath)
   at Microsoft.CodeAnalysis.Tools.FormatCommandCommon.FormatAsync(FormatOptions formatOptions, ILogger`1 logger, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Tools.Commands.RootFormatCommand.FormatCommandDefaultHandler.InvokeAsync(ParseResult parseResult, CancellationToken cancellationToken)
   at System.CommandLine.Invocation.InvocationPipeline.InvokeAsync(ParseResult parseResult, CancellationToken cancellationToken)
Error: Process completed with exit code 1.
```

### CODE REVIEW

File: csharp/Ice/async/Server.cs [hunk 1]
```diff
@@ -13,37 +8,33 @@ public static int Main(string[] args)
...
-                        adapter.add(new HelloI(cts), Ice.Util.stringToIdentity("hello"));
-                        adapter.activate();
+                Ice.ObjectAdapter adapter = communicator.createObjectAdapter("Hello");
+                adapter.add(new HelloI(cts), Ice.Util.stringToIdentity("hello"));
+                adapter.activate();
 
-                        // cts is canceled by Ctrl+C or a shutdown request.
-                        // With C# 7.1 and up, you should make Main async and call: await Task.Delay(-1, cts.Token)
-                        cts.Token.WaitHandle.WaitOne();
+                // cts is canceled by Ctrl+C or a shutdown request.
```
Comment (pepone) [inline csharp/Ice/async/Server.cs:30]: We should add a TODO or issue to fix this C# 7.1 is quite old.

**Review by pepone — COMMENTED** [review]
Looks good, but seems there is still a few warnings to fix, making CI fail

**Comment by pepone** [comment]
```
Run dotnet format --verify-no-changes "csharp/C# NET demos.sln"
Unhandled exception: System.Exception: Restore operation failed.
```

Probably because we didn't install the 3.8.0-alpha package, and it cannot be found in any of the default repositories.

File: csharp/Ice/async/Server.cs (hunk 1 above)
Comment (externl) [reply csharp/Ice/async/Server.cs:30]: I opened an issue

**Pull request merged**